src/Fixer/ClassNotation/SingleClassElementPerStatementFixer.php

```diff
@@ -0,0 +1,176 @@
+ * @author SpacePossum
+ */
+final class SingleClassElementPerStatementFixer extends AbstractFixer
+{
+    /**
+     * @var string[]
+     */
+    private $configuration;
+
+    /**
+     * Default target/configuration.
+     *
+     * @var string[]
+     */
+    private static $defaultConfiguration = array(
+        'property',
+        'const',
+    );
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        if (null === $configuration) {
+            $this->configuration = self::$defaultConfiguration;
+
+            return;
+        }
+
+        foreach ($configuration as $name) {
+            if (!in_array($name, self::$defaultConfiguration, true)) {
+                throw new InvalidFixerConfigurationException($this->getName(), sprintf('Unknown configuration option "%s".', $name));
+            }
+        }
+
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $analyzer = new TokensAnalyzer($tokens);
+        $elements = array_reverse($analyzer->getClassyElements(), true);
+
+        foreach ($elements as $index => $element) {
+            if (!in_array($element['type'], $this->configuration, true)) {
+                continue; // not in configuration
+            }
+
+            $this->fixElement($tokens, $index);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'There MUST NOT be more than one property or constant declared per statement.';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound(Token::getClassyTokenKinds());
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     */
+    private function fixElement(Tokens $tokens, $index)
+    {
+        $repeatIndex = $tokens->getNextTokenOfKind($index, array(',', ';'));
+        if (!$tokens[$repeatIndex]->equals(',')) {
+            return; // no repeating found, no fixing needed
+        }
+
+        $start = $tokens->getPrevTokenOfKind($index, array(';', '{', '}'));
+        $this->expandElement(
+            $tokens,
+            $tokens->getNextMeaningfulToken($start),
+            $tokens->getNextTokenOfKind($index, array(';'))
+        );
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $startIndex
+     * @param int    $endIndex
+     */
+    private function expandElement(Tokens $tokens, $startIndex, $endIndex)
+    {
+        // collect modifiers
+        $modifiers = $this->getModifiersSequences($tokens, $startIndex, $endIndex);
+        $divisionContent = null;
+        if ($tokens[$startIndex - 1]->isWhitespace()) {
+            $divisionContent = $tokens[$startIndex - 1]->getContent();
+            if (preg_match('#(\n|\n\r)#', $divisionContent, $matches)) {
```

Comment:
The sequence is typically `\r\n` IIRC, but it shouldn't matter, i.e. both cases should match.
So I think the match should be something like `(` 0-N times `\r` 1-N times `\n` 0-N times `\r` `)`.